### ISSUE TITLE
bazel run //exmaples/cpp:simple_psi 出错 (Mac M1)

### ISSUE THREAD

**Issue by zouhuan1215** [secretflow/spu#170]
### Issue Type

Bug

### Modules Involved

PSI

### Have you reproduced the bug with SPU HEAD?

Yes

### Installation Kind

source

### SPU Version

0.3.3b0 (the latest version)

### OS Platform and Distribution

macOS 12.5 Monterey with M1 pro

### Python Version

3.8.16 (using anaconda)

### Compiler Version

apple clang version 14.0.0 (arm64-apple-darwin21.6.0)

### Current Behavior?

When running `bazel run //examples/cpp:simple_psi -- -rank 0 -protocol 1 -in_path examples/data/psi_1.csv -field_names id -out_path /tmp/p1.out`, the code was built successfully. But the generated executable failed to run. 

### Standalone code to reproduce the issue

```Python
Run the command in the terminal `bazel run //examples/cpp:simple_psi -- -rank 0 -protocol 1 -in_path examples/data/psi_1.csv -field_names id -out_path /tmp/p1.out`
```

### Relevant log output

```Shell
INFO: Elapsed time: 262.202s, Critical Path: 110.61s
INFO: 1782 processes: 293 internal, 1489 darwin-sandbox.
INFO: Build completed successfully, 1782 total actions
INFO: Running command line: bazel-bin/examples/cpp/simple_psi -rank 0 -protoc
INFO: Build completed successfully, 1782 total actions
W0420 11:25:56   259 external/com_github_brpc_brpc/src/bvar/default_variables.cpp:453] Fail to proc_pid_rusage: Invalid argument
I0420 11:25:56   259 external/com_github_brpc_brpc/src/brpc/server.cpp:1107] Server[yacl::link::internal::ReceiverServiceImpl] is serving on port=9530.
I0420 11:25:56   259 external/com_github_brpc_brpc/src/brpc/server.cpp:1110] Check out http://FanFandeMBP.lan:9530 in web browser.
[2023-04-20 11:25:56.779] [error] [context.cc:132] connect to rank=1 failed with error [external/yacl/yacl/link/transport/channel_brpc.cc:369] send, rpc failed=64, message=[E61]Fail to connect Socket{id=0 addr=127.0.0.1:9531} (0x0x1482b8000): Connection refused [R1][E64]Not connected to 127.0.0.1:9531 yet, server_id=0 [R2][E64]Not connected to 127.0.0.1:9531 yet, server_id=0 [R3][E64]Not connected to 127.0.0.1:9531 yet, server_id=0
I0420 11:25:56  5891 external/com_github_brpc_brpc/src/brpc/socket.cpp:2344] Checking Socket{id=0 addr=127.0.0.1:9531} (0x1482b8000)
F0420 11:25:56 10499 external/com_github_brpc_brpc/src/bthread/task_group.cpp:629] bthread=4294969088 sched_to itself!
I0420 11:25:56 10499 external/com_github_brpc_brpc/src/brpc/socket.cpp:2404] Revived Socket{id=0 addr=127.0.0.1:9531} (0x1482b8000) (Connectable)
F0420 11:25:56 10499 external/com_github_brpc_brpc/src/bthread/task_group.cpp:629] bthread=4294969088 sched_to itself!
```

**Comment by zhanglei486** [comment]
Reference psi doc:
https://github.com/secretflow/spu/blob/main/docs/development/psi.rst#run-streaming-python-psi-example

Start two terminals, one as psi receiver(rank 0) run simple_psi, another one as psi sender(rank 1) run simple_psi.

**Comment by zouhuan1215** [comment]
I switched to an AMD64 ubuntu22.04 machine, then all the PSI examples of cpp and python ran as expected.

On the Macos M1 machine, when built as `release` mode, both cpp and python worked as expected. When built as `debug` mode, the error would occur which said the third-party package `brpc` (required by `libspu`) occurred a "sched_to_itself" error. When I modified the compilation option for the debug mode from `-O0` to `-O2`, the debug mode worked as expected. 

Therefore,  I think this error is platform specific and is introduced by the third-party `brpc` package. A related issue in `brpc` can be found at [ "sched_to itself" error when building by Clang on Linux aarch64 ](https://github.com/apache/brpc/issues/845)
